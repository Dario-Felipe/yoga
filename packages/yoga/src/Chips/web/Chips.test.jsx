import React from 'react';
import { render } from '@testing-library/react';

import { ThemeProvider, Chips } from '../..';
import { MapPin, ChevronDown } from '@gympass/yoga-icons';

describe('<Chips />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(
        <ThemeProvider>
          <Chips>Classes</Chips>
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with disabled prop', () => {
      const { container } = render(
        <ThemeProvider>
          <Chips disabled>Classes</Chips>
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with more than one Chip', () => {
      const { container } = render(
        <ThemeProvider>
          <Chips>Classes</Chips>
          <Chips>Gyms and studios</Chips>
          <Chips>Personal trainers</Chips>
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    describe('selected', () => {
      it('should match snapshot', () => {
        const { container } = render(
          <ThemeProvider>
            <Chips selected>Classes</Chips>
          </ThemeProvider>,
        );

        expect(container).toMatchSnapshot();
      });

      it('should match snapshot with more than one Chip', () => {
        const { container } = render(
          <ThemeProvider>
            <Chips>Classes</Chips>
            <Chips selected>Gyms and studios</Chips>
            <Chips selected>Personal trainers</Chips>
          </ThemeProvider>,
        );

        expect(container).toMatchSnapshot();
      });
    });

    describe('counter', () => {
      it('should match snapshot', () => {
        const { container } = render(
          <ThemeProvider>
            <Chips counter={8}>Classes</Chips>
          </ThemeProvider>,
        );

        expect(container).toMatchSnapshot();
      });

      it('should match snapshot with more than one Chip', () => {
        const { container } = render(
          <ThemeProvider>
            <Chips counter={8}>Classes</Chips>
            <Chips counter={52} selected>
              Gyms and studios
            </Chips>
            <Chips counter={1732} selected>
              Personal trainers
            </Chips>
          </ThemeProvider>,
        );

        expect(container).toMatchSnapshot();
      });
    });

    describe('icon', () => {
      it('should match selected snapshot', () => {
        const { container } = render(
          <ThemeProvider>
            <Chips counter={8} icon={[MapPin]}>
              Location
            </Chips>
            <Chips counter={8} icon={[MapPin, ChevronDown]}>
              Classes
            </Chips>
            <Chips icon={[MapPin, ChevronDown]}>Gyms and studios</Chips>
            <Chips icon={[MapPin]}>Personal trainers</Chips>
            <Chips counter={8} selected icon={[MapPin]}>
              Activities
            </Chips>
            <Chips counter={8} selected icon={[MapPin, ChevronDown]}>
              Language
            </Chips>
            <Chips selected icon={[MapPin, ChevronDown]}>
              Categories
            </Chips>
            <Chips selected icon={[MapPin]}>
              Availability
            </Chips>
          </ThemeProvider>,
        );

        expect(container).toMatchSnapshot();
      });
    });
  });
});
