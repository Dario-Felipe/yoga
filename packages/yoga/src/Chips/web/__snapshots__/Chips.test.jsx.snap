// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chips /> Snapshots counter should match snapshot 1`] = `
.c1 {
  margin: 0;
  padding: 0;
  font-size: 12px;
  line-height: 16px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
}

.c0 {
  height: 32px;
  max-width: 216px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-style: solid;
  cursor: pointer;
  padding: 8px 12px;
  border-radius: 9999px;
  border-width: 1px;
  border-color: rgba(35,27,34,0.4);
  background-color: #FFFFFF;
}

.c0:not(:last-child) {
  margin-right: 8px;
}

.c0[disabled] {
  background-color: #F5F5FA;
  border-color: #F5F5FA;
  cursor: not-allowed;
}

.c0[disabled] p {
  color: #D7D7E0;
}

.c0[disabled] svg {
  fill: #D7D7E0;
}

.c0 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0 p {
  color: #231B22;
}

.c0:hover:enabled {
  border-color: #231B22;
}

.c0:active:enabled {
  border-color: rgba(35,27,34,0.6);
}

.c0:active:enabled p {
  color: rgba(35,27,34,0.6);
}

.c0:active:enabled svg {
  fill: rgba(35,27,34,0.6);
}

<div>
  <button
    class="c0"
    type="button"
  >
    <p
      class="c1"
    >
      Classes
    </p>
  </button>
</div>
`;

exports[`<Chips /> Snapshots counter should match snapshot with more than one Chip 1`] = `
.c1 {
  margin: 0;
  padding: 0;
  font-size: 12px;
  line-height: 16px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
}

.c3 {
  margin: 0;
  padding: 0;
  font-size: 12px;
  line-height: 16px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 500;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 16px;
  min-width: 16px;
  padding-right: 5px;
  padding-left: 5px;
  margin-left: 4px;
  color: #231B22;
  font-size: 12px;
  line-height: 16px;
  font-weight: 500px;
  border-radius: 8px;
  background-color: #FFFFFF;
}

.c0 {
  height: 32px;
  max-width: 216px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-style: solid;
  cursor: pointer;
  padding: 8px 12px;
  border-radius: 9999px;
  border-width: 1px;
  border-color: rgba(35,27,34,0.4);
  background-color: #FFFFFF;
}

.c0:not(:last-child) {
  margin-right: 8px;
}

.c0[disabled] {
  background-color: #F5F5FA;
  border-color: #F5F5FA;
  cursor: not-allowed;
}

.c0[disabled] p {
  color: #D7D7E0;
}

.c0[disabled] svg {
  fill: #D7D7E0;
}

.c0 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0 p {
  color: #231B22;
}

.c0:hover:enabled {
  border-color: #231B22;
}

.c0:active:enabled {
  border-color: rgba(35,27,34,0.6);
}

.c0:active:enabled p {
  color: rgba(35,27,34,0.6);
}

.c0:active:enabled svg {
  fill: rgba(35,27,34,0.6);
}

.c2 {
  height: 32px;
  max-width: 216px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-style: solid;
  cursor: pointer;
  padding: 8px 12px;
  border-radius: 9999px;
  border-width: 1px;
  background-color: #231B22;
  border-color: transparent;
}

.c2:not(:last-child) {
  margin-right: 8px;
}

.c2[disabled] {
  background-color: #F5F5FA;
  border-color: #F5F5FA;
  cursor: not-allowed;
}

.c2[disabled] p {
  color: #D7D7E0;
}

.c2[disabled] svg {
  fill: #D7D7E0;
}

.c2 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2 p {
  color: #FFFFFF;
}

.c2:hover:enabled {
  border-color: #231B22;
}

.c2:active:enabled {
  background-color: rgba(35,27,34,0.6);
  border-color: transparent;
}

<div>
  <button
    class="c0"
    type="button"
  >
    <p
      class="c1"
    >
      Classes
    </p>
  </button>
  <button
    class="c2"
    type="button"
  >
    <p
      class="c3"
    >
      Gyms and studios
    </p>
    <span
      class="c4"
    >
      52
    </span>
  </button>
  <button
    class="c2"
    type="button"
  >
    <p
      class="c3"
    >
      Personal trainers
    </p>
    <span
      class="c4"
    >
      +999
    </span>
  </button>
</div>
`;

exports[`<Chips /> Snapshots icon should match selected snapshot 1`] = `
.c1 {
  margin: 0;
  padding: 0;
  font-size: 12px;
  line-height: 16px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
}

.c3 {
  margin: 0;
  padding: 0;
  font-size: 12px;
  line-height: 16px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 500;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 16px;
  min-width: 16px;
  padding-right: 5px;
  padding-left: 5px;
  margin-left: 4px;
  color: #231B22;
  font-size: 12px;
  line-height: 16px;
  font-weight: 500px;
  border-radius: 8px;
  background-color: #FFFFFF;
}

.c0 {
  height: 32px;
  max-width: 216px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-style: solid;
  cursor: pointer;
  padding: 8px 12px;
  border-radius: 9999px;
  border-width: 1px;
  border-color: rgba(35,27,34,0.4);
  background-color: #FFFFFF;
}

.c0:not(:last-child) {
  margin-right: 8px;
}

.c0[disabled] {
  background-color: #F5F5FA;
  border-color: #F5F5FA;
  cursor: not-allowed;
}

.c0[disabled] p {
  color: #D7D7E0;
}

.c0[disabled] svg {
  fill: #D7D7E0;
}

.c0 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0 p {
  color: #231B22;
}

.c0:hover:enabled {
  border-color: #231B22;
}

.c0:active:enabled {
  border-color: rgba(35,27,34,0.6);
}

.c0:active:enabled p {
  color: rgba(35,27,34,0.6);
}

.c0:active:enabled svg {
  fill: rgba(35,27,34,0.6);
}

.c2 {
  height: 32px;
  max-width: 216px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-style: solid;
  cursor: pointer;
  padding: 8px 12px;
  border-radius: 9999px;
  border-width: 1px;
  background-color: #231B22;
  border-color: transparent;
}

.c2:not(:last-child) {
  margin-right: 8px;
}

.c2[disabled] {
  background-color: #F5F5FA;
  border-color: #F5F5FA;
  cursor: not-allowed;
}

.c2[disabled] p {
  color: #D7D7E0;
}

.c2[disabled] svg {
  fill: #D7D7E0;
}

.c2 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2 p {
  color: #FFFFFF;
}

.c2:hover:enabled {
  border-color: #231B22;
}

.c2:active:enabled {
  background-color: rgba(35,27,34,0.6);
  border-color: transparent;
}

<div>
  <button
    class="c0"
    type="button"
  >
    <p
      class="c1"
    >
      Location
    </p>
  </button>
  <button
    class="c0"
    type="button"
  >
    <p
      class="c1"
    >
      Classes
    </p>
  </button>
  <button
    class="c0"
    type="button"
  >
    <p
      class="c1"
    >
      Gyms and studios
    </p>
  </button>
  <button
    class="c0"
    type="button"
  >
    <p
      class="c1"
    >
      Personal trainers
    </p>
  </button>
  <button
    class="c2"
    type="button"
  >
    <p
      class="c3"
    >
      Activities
    </p>
    <span
      class="c4"
    >
      8
    </span>
  </button>
  <button
    class="c2"
    type="button"
  >
    <p
      class="c3"
    >
      Language
    </p>
    <span
      class="c4"
    >
      8
    </span>
  </button>
  <button
    class="c2"
    type="button"
  >
    <p
      class="c3"
    >
      Categories
    </p>
  </button>
  <button
    class="c2"
    type="button"
  >
    <p
      class="c3"
    >
      Availability
    </p>
  </button>
</div>
`;

exports[`<Chips /> Snapshots selected should match snapshot 1`] = `
.c1 {
  margin: 0;
  padding: 0;
  font-size: 12px;
  line-height: 16px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 500;
}

.c0 {
  height: 32px;
  max-width: 216px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-style: solid;
  cursor: pointer;
  padding: 8px 12px;
  border-radius: 9999px;
  border-width: 1px;
  background-color: #231B22;
  border-color: transparent;
}

.c0:not(:last-child) {
  margin-right: 8px;
}

.c0[disabled] {
  background-color: #F5F5FA;
  border-color: #F5F5FA;
  cursor: not-allowed;
}

.c0[disabled] p {
  color: #D7D7E0;
}

.c0[disabled] svg {
  fill: #D7D7E0;
}

.c0 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0 p {
  color: #FFFFFF;
}

.c0:hover:enabled {
  border-color: #231B22;
}

.c0:active:enabled {
  background-color: rgba(35,27,34,0.6);
  border-color: transparent;
}

<div>
  <button
    class="c0"
    type="button"
  >
    <p
      class="c1"
    >
      Classes
    </p>
  </button>
</div>
`;

exports[`<Chips /> Snapshots selected should match snapshot with more than one Chip 1`] = `
.c1 {
  margin: 0;
  padding: 0;
  font-size: 12px;
  line-height: 16px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
}

.c3 {
  margin: 0;
  padding: 0;
  font-size: 12px;
  line-height: 16px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 500;
}

.c0 {
  height: 32px;
  max-width: 216px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-style: solid;
  cursor: pointer;
  padding: 8px 12px;
  border-radius: 9999px;
  border-width: 1px;
  border-color: rgba(35,27,34,0.4);
  background-color: #FFFFFF;
}

.c0:not(:last-child) {
  margin-right: 8px;
}

.c0[disabled] {
  background-color: #F5F5FA;
  border-color: #F5F5FA;
  cursor: not-allowed;
}

.c0[disabled] p {
  color: #D7D7E0;
}

.c0[disabled] svg {
  fill: #D7D7E0;
}

.c0 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0 p {
  color: #231B22;
}

.c0:hover:enabled {
  border-color: #231B22;
}

.c0:active:enabled {
  border-color: rgba(35,27,34,0.6);
}

.c0:active:enabled p {
  color: rgba(35,27,34,0.6);
}

.c0:active:enabled svg {
  fill: rgba(35,27,34,0.6);
}

.c2 {
  height: 32px;
  max-width: 216px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-style: solid;
  cursor: pointer;
  padding: 8px 12px;
  border-radius: 9999px;
  border-width: 1px;
  background-color: #231B22;
  border-color: transparent;
}

.c2:not(:last-child) {
  margin-right: 8px;
}

.c2[disabled] {
  background-color: #F5F5FA;
  border-color: #F5F5FA;
  cursor: not-allowed;
}

.c2[disabled] p {
  color: #D7D7E0;
}

.c2[disabled] svg {
  fill: #D7D7E0;
}

.c2 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2 p {
  color: #FFFFFF;
}

.c2:hover:enabled {
  border-color: #231B22;
}

.c2:active:enabled {
  background-color: rgba(35,27,34,0.6);
  border-color: transparent;
}

<div>
  <button
    class="c0"
    type="button"
  >
    <p
      class="c1"
    >
      Classes
    </p>
  </button>
  <button
    class="c2"
    type="button"
  >
    <p
      class="c3"
    >
      Gyms and studios
    </p>
  </button>
  <button
    class="c2"
    type="button"
  >
    <p
      class="c3"
    >
      Personal trainers
    </p>
  </button>
</div>
`;

exports[`<Chips /> Snapshots should match snapshot 1`] = `
.c1 {
  margin: 0;
  padding: 0;
  font-size: 12px;
  line-height: 16px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
}

.c0 {
  height: 32px;
  max-width: 216px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-style: solid;
  cursor: pointer;
  padding: 8px 12px;
  border-radius: 9999px;
  border-width: 1px;
  border-color: rgba(35,27,34,0.4);
  background-color: #FFFFFF;
}

.c0:not(:last-child) {
  margin-right: 8px;
}

.c0[disabled] {
  background-color: #F5F5FA;
  border-color: #F5F5FA;
  cursor: not-allowed;
}

.c0[disabled] p {
  color: #D7D7E0;
}

.c0[disabled] svg {
  fill: #D7D7E0;
}

.c0 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0 p {
  color: #231B22;
}

.c0:hover:enabled {
  border-color: #231B22;
}

.c0:active:enabled {
  border-color: rgba(35,27,34,0.6);
}

.c0:active:enabled p {
  color: rgba(35,27,34,0.6);
}

.c0:active:enabled svg {
  fill: rgba(35,27,34,0.6);
}

<div>
  <button
    class="c0"
    type="button"
  >
    <p
      class="c1"
    >
      Classes
    </p>
  </button>
</div>
`;

exports[`<Chips /> Snapshots should match snapshot with disabled prop 1`] = `
.c1 {
  margin: 0;
  padding: 0;
  font-size: 12px;
  line-height: 16px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 500;
}

.c0 {
  height: 32px;
  max-width: 216px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-style: solid;
  cursor: pointer;
  padding: 8px 12px;
  border-radius: 9999px;
  border-width: 1px;
  border-color: rgba(35,27,34,0.4);
  background-color: #FFFFFF;
}

.c0:not(:last-child) {
  margin-right: 8px;
}

.c0[disabled] {
  background-color: #F5F5FA;
  border-color: #F5F5FA;
  cursor: not-allowed;
}

.c0[disabled] p {
  color: #D7D7E0;
}

.c0[disabled] svg {
  fill: #D7D7E0;
}

.c0 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0 p {
  color: #231B22;
}

.c0:hover:enabled {
  border-color: #231B22;
}

.c0:active:enabled {
  border-color: rgba(35,27,34,0.6);
}

.c0:active:enabled p {
  color: rgba(35,27,34,0.6);
}

.c0:active:enabled svg {
  fill: rgba(35,27,34,0.6);
}

<div>
  <button
    class="c0"
    disabled=""
    type="button"
  >
    <p
      class="c1"
    >
      Classes
    </p>
  </button>
</div>
`;

exports[`<Chips /> Snapshots should match snapshot with more than one Chip 1`] = `
.c1 {
  margin: 0;
  padding: 0;
  font-size: 12px;
  line-height: 16px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
}

.c0 {
  height: 32px;
  max-width: 216px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-style: solid;
  cursor: pointer;
  padding: 8px 12px;
  border-radius: 9999px;
  border-width: 1px;
  border-color: rgba(35,27,34,0.4);
  background-color: #FFFFFF;
}

.c0:not(:last-child) {
  margin-right: 8px;
}

.c0[disabled] {
  background-color: #F5F5FA;
  border-color: #F5F5FA;
  cursor: not-allowed;
}

.c0[disabled] p {
  color: #D7D7E0;
}

.c0[disabled] svg {
  fill: #D7D7E0;
}

.c0 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0 p {
  color: #231B22;
}

.c0:hover:enabled {
  border-color: #231B22;
}

.c0:active:enabled {
  border-color: rgba(35,27,34,0.6);
}

.c0:active:enabled p {
  color: rgba(35,27,34,0.6);
}

.c0:active:enabled svg {
  fill: rgba(35,27,34,0.6);
}

<div>
  <button
    class="c0"
    type="button"
  >
    <p
      class="c1"
    >
      Classes
    </p>
  </button>
  <button
    class="c0"
    type="button"
  >
    <p
      class="c1"
    >
      Gyms and studios
    </p>
  </button>
  <button
    class="c0"
    type="button"
  >
    <p
      class="c1"
    >
      Personal trainers
    </p>
  </button>
</div>
`;
